src/decode.h

```diff
@@ -307,3 +300,5 @@ typedef struct rv_insn {
 
 /* decode the RISC-V instruction */
 bool rv_decode(rv_insn_t *ir, const uint32_t insn);
+
+#define CHECK_COMPRESSED_INSN ((insn & FC_OPCODE) != 3)
```

Comment:
Rewrite as an inline function, so that compiler can check the types. In addition, write the corresponding comment for the detection of RV32C instructions.